### PR TITLE
Roll to Playwright v1.41.1

### PR DESCRIPTION
**Pull request by canstand**
In addition, the last added `route` is matched first, which is consistent with the upstream.

### CODE REVIEW

File: browser_context.go [hunk 1]
```diff
@@ -210,21 +214,57 @@ func (b *browserContextImpl) ExposeFunction(name string, binding ExposedFunction
...
+}
+
+func (b *browserContextImpl) unrouteInternal(removed []*routeHandlerEntry, remaining []*routeHandlerEntry, behavior *UnrouteBehavior) error {
 	b.Lock()
 	defer b.Unlock()
-
-	routes, err := unroute(b.routes, url, handlers...)
+	b.routes = remaining
+	err := b.updateInterceptionPatterns()
 	if err != nil {
```
Comment (mxschmitt) [inline browser_context.go:236]:
```suggestion
	if err := b.updateInterceptionPatterns(); err != nil {
```

**Review by mxschmitt — APPROVED** [review]
Great work!

**Review by mxschmitt — APPROVED** [review]

File: browser_context.go (hunk 1 above)
Comment (canstand) [reply browser_context.go]: Hi @mxschmitt , I'm considering forcing the use of `gofumpt` to format the code, which would avoid these. What do you think?

Comment (mxschmitt) [reply browser_context.go]: Feel free to add!

**Pull request merged**